extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/client/VertxRedisClientFactory.java

```diff
@@ -99,6 +101,21 @@ public static Redis create(String name, Vertx vertx, RedisClientConfig config) {
         return Redis.createClient(vertx, options);
     }
 
+    private static URI addClientParameter(String clientName, URI uri) {
+        try {
+            String query = uri.getQuery() == null ? "client=%s".formatted(clientName)
```

Comment:
We would need to do a lot more tests because of the infamous URL encoding:

- what about -, _, $, % and other funky character
- what about spaces

Reply:
+1000, this is super important.